src/frontend/src/components/toast.ts

```diff
@@ -1,60 +1,67 @@
+import { closeIcon, warningIcon } from "./icons";
 
-export const removeToast = (toast: HTMLElement): void => {
-  toast.addEventListener("animationend", () => {
-    toast.remove();
-  });
-  toast.classList.add("c-toast--closing");
-};
-
-const createToast = (message: TemplateElement): HTMLDivElement => {
-  const toastEl = document.createElement("div");
-  toastEl.classList.add("c-toast");
-  render(
-    irregularity({
-      message,
-      closeFn: () => removeToast(toastEl),
-    }),
-    toastEl
-  );
-  return toastEl;
-};
-
-/**
- * Get the container for the toasts, or create it if it doesn't exist.
- * @returns the container reference for the toasts
- */
-const getOrCreateToaster = (): Element | null => {
-  const toasterEl = document.querySelector("[data-toaster]");
-  if (!toasterEl) {
-    // generic toaster that will be appended to the body
-    const toasterEl = document.createElement("div");
-    toasterEl.classList.add("c-toasts");
-    // container for the toasts. Two reasons for this:
-    // 1. we can reuse the l-container class for styling
-    // 2. the toasts will only be as wide as the container (user can still interact with the rest of the page)
-    const containerEl = document.createElement("div");
-    containerEl.classList.add("l-container");
-    // Attribute used to retrieve the element to append toasts to
-    containerEl.setAttribute("data-toasts", "");
-    toasterEl.setAttribute("data-toaster", "");
-    toasterEl.appendChild(containerEl);
-    document.body.appendChild(toasterEl);
-    return containerEl;
-  }
-  return toasterEl.querySelector("[data-toasts]");
-};
+// A toast element containing a message, and the (static) list of all toasts. The `Toast` element
+// must be an object because we (lit) use object-references in `repeat` to figure out which DOM elements
+// to replace.
+type Toast = { message: TemplateElement };
+const toasts: Toast[] = []; // the "toasts" singleton
 
 /**
  * A lot like console.error, but with a toast.
  * use like: toast.error("Something went wrong");
  */
 export const toast = {
-  error: (message: TemplateElement): void => {
-    const toasterEl = getOrCreateToaster();
-    const toastEl = createToast(message);
-    toasterEl?.appendChild(toastEl);
+  error: (message: Toast["message"]): void => {
+    toasts.push({ message });
+    renderToasts();
   },
 };
+
+// Render or rerender toasts.
+const renderToasts = () => {
+  render(
+    html` <div class="c-toasts l-container">
+      ${repeat(toasts, (x) => x, toastTemplate)}
+    </div>`,
+    document.body
+  );
+};
+
+const toastTemplate = (toast: Toast): TemplateResult => {
+  const closing = new Chan<boolean>(false);
+  const closeToast = () => closing.send(true);
+
+  const removeToast = () => {
+    const toastIndex = toasts.indexOf(toast);
+    if (toastIndex > -1) {
+      toasts.splice(toastIndex, 1);
+    }
+    renderToasts();
+  };
+
+  return html` <div
+    class="c-toast ${asyncReplace(
```

Comment:
I think we could use

```
role="alert" aria-live="assertive" 
```

So user using screenreaders would have the error read out loud to them the moment it is triggered.

Reply:
Added to the todo list! I'll do this separately, I've tried not to change anything in the generated HTML (except for squishing the divs to simplify)